cpp/webcrypto/crypto_ec.cpp

```diff
@@ -331,4 +331,110 @@ jsi::Value GetEcKeyDetail(jsi::Runtime &rt,
   return target;
 }
 
+EcKeyPairGenConfig prepareEcKeyGenConfig(jsi::Runtime &rt,
+                                       const jsi::Value *args)
+{
+  EcKeyPairGenConfig config = EcKeyPairGenConfig();
+
+  // curve name
+  std::string curveName = args[1].asString(rt).utf8(rt);
+  config.curve_nid = GetCurveFromName(curveName.c_str());
+
+  // encoding
+  if (CheckIsInt32(args[2].asNumber())) {
+    int encoding = static_cast<int>(args[2].asNumber());
+    if (encoding != OPENSSL_EC_NAMED_CURVE &&
+        encoding != OPENSSL_EC_EXPLICIT_CURVE) {
+      throw jsi::JSError(rt, "Invalid param_encoding specified");
+    } else {
+      config.param_encoding = encoding;
+    }
+  } else {
+    throw jsi::JSError(rt, "Invalid param_encoding specified (not int)");
+  }
+
+  // rest of args for encoding
+  unsigned int offset = 3;
+
+  config.public_key_encoding = ManagedEVPPKey::GetPublicKeyEncodingFromJs(
+      rt, args, &offset, kKeyContextGenerate);
+
+  auto private_key_encoding = ManagedEVPPKey::GetPrivateKeyEncodingFromJs(
+      rt, args, &offset, kKeyContextGenerate);
+
+  if (!private_key_encoding.IsEmpty()) {
+    config.private_key_encoding = private_key_encoding.Release();
+  }
+
+  return config;
+}
+
+EVPKeyCtxPointer setup(std::shared_ptr<EcKeyPairGenConfig> config) {
+  EVPKeyCtxPointer key_ctx;
+  switch (config->curve_nid) {
+    case EVP_PKEY_ED25519:
+      // Fall through
+    case EVP_PKEY_ED448:
+      // Fall through
+    case EVP_PKEY_X25519:
+      // Fall through
+    case EVP_PKEY_X448:
+      key_ctx.reset(EVP_PKEY_CTX_new_id(config->curve_nid, nullptr));
+      break;
+    default: {
+      EVPKeyCtxPointer param_ctx(EVP_PKEY_CTX_new_id(EVP_PKEY_EC, nullptr));
+      EVP_PKEY* raw_params = nullptr;
+      if (!param_ctx ||
+          EVP_PKEY_paramgen_init(param_ctx.get()) <= 0 ||
+          EVP_PKEY_CTX_set_ec_paramgen_curve_nid(
+              param_ctx.get(), config->curve_nid) <= 0 ||
+          EVP_PKEY_CTX_set_ec_param_enc(
+              param_ctx.get(), config->param_encoding) <= 0 ||
+          EVP_PKEY_paramgen(param_ctx.get(), &raw_params) <= 0) {
+        return EVPKeyCtxPointer();
+      }
+      EVPKeyPointer key_params(raw_params);
+      key_ctx.reset(EVP_PKEY_CTX_new(key_params.get(), nullptr));
+    }
+  }
+
+  if (key_ctx && EVP_PKEY_keygen_init(key_ctx.get()) <= 0)
+    key_ctx.reset();
+
+  return key_ctx;
+}
+
+std::pair<jsi::Value, jsi::Value> generateEcKeyPair(jsi::Runtime& runtime,
```

Comment:
I like that we go away from `JSVariant`, but I wonder if it makes sense to use C++ types instead of `jsi::Value` here.

If we do that, we could avoid Runtime specific calls and allow this method to run on any Thread.

Since we use `jsi::Runtime` here, we can only call this method on the JS-thread. So this cannot be called async.